Services/UserService.Vector.cs

```diff
@@ -0,0 +1,127 @@
+using Tokengram.Database.Tokengram.Entities;
+using Tokengram.Services.Interfaces;
+using Tokengram.Constants;
+using Tokengram.Database.Neo4j.Nodes;
+
+namespace Tokengram.Services
+{
+    public partial class UserService : IUserService
+    {
+        public async Task<User> UpdateUserVectorByAction(User user, Post post, int actionCoefficient)
+        {
+            var nft = await _nftService.GetNFT(post.NFTAddress);
+            var nftVector = nft?.NFT?.NFTVector;
+            if (nftVector == null)
+                return user;
+
+            user.UserVector = GetNewVector(user.UserVector, nftVector, actionCoefficient);
+
+            await _dbContext.SaveChangesAsync();
+
+            return user;
+        }
+
+        public async Task<User> UpdateUserVectorByLike(User user, Post post)
+        {
+            return await UpdateUserVectorByAction(user, post, UpdatingVectorCoefficients.LIKE_COEFFICIENT);
+        }
+
+        public async Task<User> UpdateUserVectorByUnlike(User user, Post post)
+        {
+            return await UpdateUserVectorByAction(user, post, -1 * UpdatingVectorCoefficients.LIKE_COEFFICIENT);
+        }
+
+        public async Task<User> UpdateUserVectorByComment(User user, Post post)
+        {
+            return await UpdateUserVectorByAction(user, post, UpdatingVectorCoefficients.COMMENT_COEFFICIENT);
+        }
+
+        public async Task<User> UpdateUserVectorByUncomment(User user, Post post)
+        {
+            return await UpdateUserVectorByAction(user, post, -1 * UpdatingVectorCoefficients.COMMENT_COEFFICIENT);
+        }
+
+        public async Task<User> UpdateUserVectorByNewNFT(User user, NFTNode nft)
+        {
+            user.UserVector = GetNewVector(
+                user.UserVector,
+                nft?.NFTVector ?? string.Empty,
+                UpdatingVectorCoefficients.NEW_NFT_COEFFICIENT
+            );
+
+            await _dbContext.SaveChangesAsync();
+            return user;
+        }
+
+        public Dictionary<string, int> ParseVectorString(string vectorString)
+        {
+            var vector = new Dictionary<string, int>();
+
+            if (string.IsNullOrEmpty(vectorString))
+                return vector;
+
+            var keyValuePairs = vectorString.Split(';');
+
+            foreach (var pair in keyValuePairs)
+            {
+                var keyValue = pair.Split(':');
+                if (keyValue.Length == 2 && int.TryParse(keyValue[1], out int value))
+                {
+                    vector[keyValue[0]] = value;
+                }
+            }
+
+            return vector;
+        }
```

Comment:
There's already very similar function `CosineSimiliarityUtil.ParseVector`. Consider using that function and possibly add parameter for specifying separator.